src/shared/parameter_v2/parameter.bzl

```diff
@@ -0,0 +1,43 @@
+def _generate_dynamic_parameters_impl(ctx):
+    output_file = ctx.outputs.generated_parameter_file
+
+    args = ctx.actions.args()
+    args.add("--output_file", output_file)
+
+    if output_file.extension == "hpp":
+        args.add("-cpp")
+    elif output_file.extension == "h":
+        args.add("-c")
+    else:
+        fail(
+            msg = "Invalid file extension. Expected 'h' or 'hpp'",
+            attr = output_file.name,
+        )
+
+    # Retrieve paths to the header files directly listed in each enum_target
+    # These paths are relative to the bazel WORKSPACE root
+    enum_headers = [f for target in ctx.attr.enum_targets for f in target[CcInfo].compilation_context.direct_headers]
```

Comment:
Just a note, `direct_headers` is only available in `bazel >= 3.1.0`. Most people are probably on `3.0.0` right now (I was). It's available in `apt` so should just be a matter of getting people to upgrade / re-run the setup scripts.